plugins/slo/pkg/slo/interfaces.go

```diff
@@ -0,0 +1,64 @@
+package slo
+
+import (
+	"context"
+
+	v1 "github.com/alexandreLamarre/oslo/pkg/manifest/v1"
+	"github.com/hashicorp/go-hclog"
+	"github.com/rancher/opni/pkg/slo/shared"
+	apis "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	sloapi "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	"google.golang.org/protobuf/proto"
+)
+
+var datasourceToImpl map[string]SLOImpl = make(map[string]SLOImpl)
+
+func init() {
+	datasourceToImpl[shared.MonitoringDatasource] = SLOMonitoring{}
+}
+
+type SLOImpl interface {
+	New(req proto.Message, p *Plugin, ctx context.Context, lg hclog.Logger) SLOImpl
+	// This method signature has to handle storage of the SLO in the KVStore
+	// Since there can be partial successes
+	createSLOImpl([]v1.SLO) (*apis.CreatedSLOs, error)
```

Comment:
these methods should probably be exported, and I would suggest renaming them like `Create` `Update` `Delete` etc.

Reply:
Done